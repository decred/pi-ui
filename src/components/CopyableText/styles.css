.copyableWrapper {
  display: flex;
  align-items: center;
  flex-wrap: nowrap;
}

.addressWrapper {
  padding: 0.3rem 1rem;
  word-break: break-all;
  background-color: var(--color-blue-lighter);
}

.copyToClipboard {
  width: 1.4rem;
  height: 1.4rem;
  cursor: pointer;
  margin-left: 5px;
  user-select: none;
}
