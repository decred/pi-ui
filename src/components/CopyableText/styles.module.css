.copyableWrapper {
  display: flex;
  align-items: center;
  flex-wrap: nowrap;
  max-width: 100%;
}

.copyToClipboard {
  width: 1.4rem;
  height: 1.4rem;
  cursor: pointer;
  margin-left: 5px;
  user-select: none;
}
