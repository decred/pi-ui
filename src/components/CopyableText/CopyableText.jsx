import PropTypes from "prop-types";
import React, { useState } from "react";
import { copyToClipboard as copy } from "./helpers";
import { classNames, idPropTypeCheckForTruncatedComponents } from "../../utils";
import styles from "./styles.css";
import Tooltip from "../Tooltip/Tooltip.jsx";
import Icon from "../Icon/Icon.jsx";
import TextHighlighted from "../TextHighlighted/TextHighlighted.jsx";

const CopyableText = ({
  id,
  truncate,
  children,
  className,
  hoverText,
  textClassName,
  textStyle,
  buttonClassName,
  buttonStyle,
  tooltipPlacement,
  ...props
}) => {
  const [feedbackActive, setFeedbackActive] = useState(false);
  const onCopyToClipboard = (value) => {
    copy(value);
    setFeedbackActive(true);
    setTimeout(() => {
      setFeedbackActive(false);
    }, 1000);
  };
  return (
    <div className={classNames(styles.copyableWrapper, className)} {...props}>
      <TextHighlighted
        id={id}
        truncate={truncate}
        className={textClassName}
        style={textStyle}>
        {children}
      </TextHighlighted>
      <Tooltip
        placement={tooltipPlacement}
        content={feedbackActive ? "Copied!" : hoverText}>
        <Icon
          type="copyToClipboard"
          backgroundColor="var(--color-primary)"
          iconColor="var(--color-gray-lightest2)"
          onClick={() => onCopyToClipboard(children)}
          className={classNames(styles.copyToClipboard, buttonClassName)}
          style={buttonStyle}
        />
      </Tooltip>
    </div>
  );
};

CopyableText.propTypes = {
  truncate: PropTypes.bool,
  children: PropTypes.node,
  hoverText: PropTypes.node,
  className: PropTypes.string,
  textClassName: PropTypes.string,
  textStyle: PropTypes.object,
  buttonClassName: PropTypes.string,
  buttonStyle: PropTypes.object,
  tooltipPlacement: PropTypes.string,
  id: idPropTypeCheckForTruncatedComponents,
};

CopyableText.defaultProps = {
  truncate: true,
  hoverText: "Copy to clipboard",
  tooltipPlacement: "right",
};

export default CopyableText;
