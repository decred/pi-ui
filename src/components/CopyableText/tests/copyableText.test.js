import React from "react";
import CopyableText from "../CopyableText";
import { create } from "react-test-renderer";
import {
  defaultLightTheme,
  ThemeProvider,
  DEFAULT_LIGHT_THEME_NAME,
} from "../../../theme";

describe("CopyableText component", () => {
  test("Matches snapshot", () => {
    const copyableText = create(
      <ThemeProvider
        themes={{ [DEFAULT_LIGHT_THEME_NAME]: defaultLightTheme }}
        defaultThemeName={DEFAULT_LIGHT_THEME_NAME}>
        <CopyableText id="copy-id">test</CopyableText>
      </ThemeProvider>
    );
    expect(copyableText.toJSON()).toMatchSnapshot();
  });
});
