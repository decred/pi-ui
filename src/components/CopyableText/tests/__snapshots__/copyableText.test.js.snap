// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyableText component Matches snapshot 1`] = `
<div
  className="copyableWrapper"
>
  <span
    className="text textAlignStart weightRegular sizeNormal colorDefault contentWrapper"
    id="copy-id"
  >
    test
  </span>
  <div
    className="tooltip"
    onClick={[Function]}
  >
    <div
      className="tooltipContent right"
    >
      Copy to clipboard
    </div>
    <svg
      className="copyToClipboard"
      height="16"
      onClick={[Function]}
      viewBox="0 0 450 450"
      width="16"
    >
      <g>
        <g>
          <g
            fill="var(--color-primary)"
            transform="translate(40.909091, 51.660000)"
          >
            <polygon
              points="306.818182 0 306.818182 67.68 61.3636364 67.68 61.3636364 0 0 0 0 338.37 368.181818 338.37 368.181818 0"
            />
          </g>
          <polygon
            fill="var(--color-primary)"
            points="136.227273 95.16 313.854545 95.16 313.854545 30.03 136.227273 30.03"
          />
          <rect
            fill="var(--color-gray-lightest2)"
            height="21.75"
            width="249.75"
            x="100.063636"
            y="165.36"
          />
          <rect
            fill="var(--color-gray-lightest2)"
            height="21.75"
            width="249.75"
            x="100.063636"
            y="233.04"
          />
          <rect
            fill="var(--color-gray-lightest2)"
            height="21.75"
            width="249.75"
            x="100.063636"
            y="300.69"
          />
        </g>
      </g>
    </svg>
  </div>
</div>
`;
